Source/Dafny/Resolver.cs

```diff
@@ -712,11 +716,17 @@ enum ValuetypeVariety {
         }
       }
 
-      foreach (var rewriter in rewriters) {
+      foreach (var rewriter in rewriters.Take(builtInRewritersCount)) {
+        rewriter.PostResolve(prog);
+      }
+      ErrorsCountWithoutPlugin = reporter.ErrorCount;
```

Comment:
This code now strongly depends on rewriters starting with resolver rewriters. Is it a concern that the order would change in the future, but someone forgetting to update this code? Is it possible to get the source of errors in another way, from the errors themselves?

Reply:
Yes. Right now, we provide plugins with an `ErrorReporter`. They can raise issues using typically `ErrorReporter.Error(MessageSource source, IToken tok, string msg)`
And, plugins would typically use `MessageSource.Resolver` as the first argument since they are plugin in as resolver issues.
If they used `MessageSource.Verifier`, currently, this message source would be ignored and the pipeline would just stop after resolution.
However, this PR opens the way that error messages using the source `MessageSource.Verifier` would not prevent the verifier from running. For that, I would remove the newly introduced property `Resolver.ErrorsCountWithoutPlugin`, and instead of `nt resolverErrors = resolver.ErrorsCountWithoutPlugin; if (resolverErrors > 0) {...` I would simply look at all non verifications errors raised by the resolver.